Cabal-tests/Cabal-tests.cabal

```diff
@@ -149,6 +149,7 @@ test-suite hackage-tests
     , Cabal
     , Cabal-syntax
     , Cabal-tree-diff
+    , cabal-install
```

Comment:
Don't pollute `Cabal` tests, with `cabal-install` stuff. 

The finding of config file logic was duplicated on purpose.

Reply:
I emphasize, this is **very** important. `Cabal` is tested with e.g. older GHCs for which in-repo `cabal-install` is not buildable.

Reply:
I guess I can bring back the duplication.  Basic CABAL_DIR support is perhaps enough for the test suite, if we can assume it is always set.

Reply:
It's not. That's why there is logic for `CABAL_CONFIG`.

Reply:
I duplicated enough logic to make this work now.